### ISSUE TITLE
Support glimpse function

### ISSUE THREAD

**Issue by chriscardillo** [machow/siuba#409]
Would be great to have dplyr's [glimpse](https://dplyr.tidyverse.org/reference/glimpse.html) function implemented in siuba and loaded in with `from siuba import *`.

`glimpse` is really useful because pandas dataframe don't always print nicely.

**Comment by machow** [comment]
Agreed it would be nice! As I understand, glimpse (which is implemented in the [pillar package](https://github.com/r-lib/pillar)) basically does the following:

* **transposes** the normal table view, so each column shows up as a row
* **prints shape** up top (e.g. Rows: 10, Columns: 2)
* for each column, **chooses the number of elements** to show based on terminal width
  - e.g. one column might show 3, while another shows 15 (shorter representations)
* has a generic 


## Examples

### Choosing n elements to show per column

```R
glimpse(
  data.frame(
    x = c(paste0(rep("abc", 10), collapse = ""), rep("zzz", 9)),
    y = 1:10
  )
)
```

<img width="578" alt="image" src="https://user-images.githubusercontent.com/2574498/160716948-cfe58846-bac7-4077-9c0b-8c4c15c1e9c7.png">

### Different from straight transposing

Transposing a dataframe doesn't dynamically select number of columns (which is much easier....!):

```python
from siuba.data import mtcars
mtcars.head().T
```

<img width="721" alt="image" src="https://user-images.githubusercontent.com/2574498/160717755-13befd29-7022-4c77-bfaf-ed5138e7af9f.png">


(Note how this also jacks up representations)

### Handles nested representations via a summary

<img width="576" alt="image" src="https://user-images.githubusercontent.com/2574498/160718103-f6ccbf66-6c02-4794-8951-30087551f90f.png">

To see how it handles each column:

```R
nested = mtcars %>% nest(data = c(-cyl))
pillar:::format_glimpse_1(nested$data)
# [1] "[<tbl_df[7 x 10]>], [<tbl_df[11 x 10]>], [<tbl_df[14 x 10]>]"
```

**Comment by pwwang** [comment]
This is cool. Also want to implement it with [datar](https://github.com/pwwang/datar).
